src/overmap_location.h

```diff
@@ -37,6 +39,9 @@ void load( JsonObject &jo, const std::string &src );
 void check_consistency();
 void reset();
 
+const std::vector<overmap_location> &get_all();
+const std::string get_random_terrain( const std::string target_location_id );
```

Comment:
You can safely get rid of these two functions if `need_id()` returns `int_id<overmap_location>`. I'd also rename `need_id()` to a more specific, location-related name.

Reply:
The idea of `get_random_terrain` function is to get random terrain from overmap location with given name.

Reply:
You don't need a name if you have an identifier that already references the desired object. Simply put: `get_location_for( need )->get_random_terrain()` will do the trick (assuming `get_location_for` is the former `need_id` which now returns `int_id<overmap_location>`). It runs in constant time instead of O(n) for the linear search in `overmap_locations::get_random_terrain`.